src/modules/admin/components/services/ServiceTypeDialog.tsx

```diff
@@ -0,0 +1,192 @@
+import { Stack } from '@mui/system';
+import * as React from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import CommonDialog from '@/components/elements/CommonDialog';
+import LabeledCheckbox from '@/components/elements/input/LabeledCheckbox';
+import TextInput from '@/components/elements/input/TextInput';
+import ValidationErrorList from '@/modules/errors/components/ValidationErrorList';
+import {
+  emptyErrorState,
+  ErrorState,
+  partitionValidations,
+} from '@/modules/errors/util';
+import CreatableFormSelect from '@/modules/form/components/CreatableFormSelect';
+import FormActions from '@/modules/form/components/FormActions';
+import { isPickListOption } from '@/modules/form/types';
+import {
+  CreateServiceTypeMutation,
+  PickListOption,
+  PickListType,
+  ServiceTypeFieldsFragment,
+  UpdateServiceTypeMutation,
+  useCreateServiceTypeMutation,
+  useGetPickListQuery,
+  useUpdateServiceTypeMutation,
+  ValidationError,
+} from '@/types/gqlTypes';
+import { evictQuery } from '@/utils/cacheUtil';
+
+interface ServiceTypeDialogProps {
+  serviceType?: ServiceTypeFieldsFragment;
+  dialogOpen: boolean;
+  closeDialog: () => void;
+}
+
+const ServiceTypeDialog: React.FC<ServiceTypeDialogProps> = ({
+  serviceType,
+  dialogOpen,
+  closeDialog,
+}) => {
+  const [name, setName] = useState(serviceType?.name || '');
+  const [supportsBulkAssignment, setSupportsBulkAssignment] = useState(
+    serviceType?.supportsBulkAssignment || false
+  );
+  const [serviceCategory, setServiceCategory] = useState<PickListOption | null>(
+    null
+  );
+  const [errors, setErrors] = useState<ErrorState>(emptyErrorState);
+
+  const mutationInput = useMemo(() => {
+    return {
+      name,
+      supportsBulkAssignment,
+      serviceCategoryId: !!serviceCategory?.label ? serviceCategory.code : null,
+      serviceCategoryName: !serviceCategory?.label
+        ? serviceCategory?.code
+        : null,
+    };
+  }, [
+    name,
+    supportsBulkAssignment,
+    serviceCategory?.label,
+    serviceCategory?.code,
+  ]);
+
+  const {
+    data: { pickList } = {},
+    loading: pickListLoading,
+    error: pickListError,
+  } = useGetPickListQuery({
+    variables: { pickListType: PickListType.CustomServiceCategories },
+  });
+
+  useEffect(() => {
+    // When serviceType exists (meaning this is an edit dialog, not a new unsaved record),
+    // populate its existing serviceCategory into the Service Category dropdown.
+    // Do this in a useEffect so that it works regardless of whether the pickList is cached or fetched from network
+    if (serviceType && pickList) {
+      // Search in the picklist for the option matching this service's category
+      const pickListOption = pickList.find(
+        (p) => p.code === serviceType.categoryRecord.id
+      );
+      // If we didn't find it, the service category is probably a HUD service category (those aren't loaded in the picklist).
```

Comment:
See https://github.com/open-path/Green-River/issues/6831#issuecomment-2435629312

Reply:
Per convo on the previous PR, I removed some of the complexity here since it is no longer needed - the pick list will now include all categories that have any custom types.